### DISCUSSION TITLE
PDM Microphone Usage on Seeed Studio XIAO nRF52840 Sense | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#347] [Q&A]
# PDM Microphone Usage on Seeed Studio XIAO nRF52840 Sense | Seeed Studio Wiki

PDM Microphone Usage on Seeed Studio XIAO nRF52840 Sense

https://wiki.seeedstudio.com/XIAO-BLE-Sense-PDM-Usage/

**Comment by safiurrrehman** [comment]
How exactly can we use XIAO BLE Sense to collect audio data from PDM microphone via Python?

**Reply by MatthewJeffson** [reply]
You got me with this :D
Maybe try it with [circuitpython](https://wiki.seeedstudio.com/XIAO-BLE_CircutPython/) first?

**Comment by cafevirtual1** [comment]
Hi: When connect the XIAO 52840 sense microphone with “edge-impulse-data-forwarder” and try to use data acquisition i get the error: Invalid data collected from device. Also on Collect data section i see Sensor: Sensor with 1 axes Frequency: 2646Hz. On the logs appear: Failed to sample data Error: Invalid data collected from device. Can you help me?

**Reply by MatthewJeffson** [reply]
Hi! Sometimes the PDM is not working properly. The voice collection using microphone on the computer is better than using digital microphone on Seeed Studio XIAO nrf52840 and brings more accuracy.
